firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FirebaseAppDistributionImpl.java

```diff
@@ -126,9 +124,18 @@ public UpdateTask updateIfNewReleaseAvailable() {
           remakeUpdateConfirmationDialog = false;
           dialogHostActivity = null;
 
-          lifecycleNotifier
-              .applyToForegroundActivityTask(this::showSignInConfirmationDialog)
-              .onSuccessTask(lightweightExecutor, unused -> signInTester())
+          signInStorage
+              .getSignInStatus()
+              .onSuccessTask(
+                  lightweightExecutor,
+                  signedIn -> {
+                    if (signedIn) {
+                      return Tasks.forResult(null);
+                    }
+                    return lifecycleNotifier
+                        .applyToForegroundActivityTask(this::showSignInConfirmationDialog)
+                        .onSuccessTask(lightweightExecutor, unused -> signInTester());
+                  })
```

Comment:
maybe extract a method named something like `signInIfNeeded()` or `ensureSignedIn()` which encapsulates this

Reply:
actually, this seems like `TesterSignInManager.signInTester()`  - could that be called from here?

Reply:
`TesterSignInManager.signInTester()` opens the actual sign in screen in the browser.

Agreed that it would be nice to extract this though, if only to make this method more readable. I called it `signInWithConfirmationIfNeeded()`.